.github/workflows/ci-linux.yml

```diff
@@ -0,0 +1,108 @@
+# Github Actions workflow for Continuous Integration testing on Linux.
+# TODO i#4549: This is a work-in-progress currently.
+
+name: ci-linux
+on: 
```

Comment:
@derekbruening Do we need to commit this incomplete config to master, so that other contributors have a skeleton config to work with? 

With this config, the two arm cross-compile builds are passing, and the two Linux tests suites have a few tests that need some work.

We can change the `on:` section to disable automatic triggers while we're working on it.

Reply:
SGTM.  I'm on vacation today so prob you'll be the only contributor until at least Monday, but there are enough other steps that I would say yes let's put this in incrementally which will make it a lot easier for others to add pieces.

> We can change the on: section to disable automatic triggers while we're working on it.

Alternatively, it could always run and we could mark the "must pass to merge" as false in the repo settings, but that might confuse people to have red tests.

Reply:
Failures in the two x86 suites went away when I changed to Xenial (see Checks for 8dfe18e). 

> Alternatively, it could always run and we could mark the "must pass to merge" as false in the repo settings, but that might confuse people to have red tests.

Yeah, it might be confusing. Better to go with disabling it for now. We can always enable PR-locally by uncommenting in the PR draft.

> there are enough other steps that I would say yes let's put this in incrementally which will make it a lot easier for others to add pieces.

I'm going to create new branch for the other steps. We can submit this one whenever possible.

Reply:
> I'm on vacation today so prob you'll be the only contributor until at least Monday

No rush on this review. Thanks.

Reply:
FYI, the manual trigger on the 'Actions' page won't show up until this PR is submitted.

https://github.community/t/workflow-files-only-picked-up-from-master/16129

Reply:
But I ran the new workflow on a forked repo: https://github.com/abhinav92003/dynamorio/runs/1444256949